main.go

```diff
@@ -114,7 +114,7 @@ func main() {
 	case "aws":
 		p, err = provider.NewAWSProvider(cfg.DomainFilter, cfg.DryRun)
 	case "inmemory":
-		p, err = provider.NewInMemoryProviderWithDomainAndLogging("example.com"), nil
+		p, err = provider.NewInMemoryProvider(provider.InMemoryWithDomain(cfg.DomainFilter), provider.InMemoryWithLogging()), nil
```

Comment:
To me this looks like this:
* I wrap arbitrary code into a function satisfying `InMemoryOption`
* Then I can pass arbitraty `InMemoryOption`s to a single `NewInMemoryProvider` function
* This allows me to modify arbitrary aspects of the target object without changing the api

I'm happy to give it a try.